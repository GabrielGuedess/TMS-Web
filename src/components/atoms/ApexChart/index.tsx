'use client';

import dynamic from 'next/dynamic';

import { type ApexChartProps } from './types';

const ReactApexChart = dynamic(() => import('react-apexcharts'), {
  ssr: false,
});

export const ApexChart = ({
  type,
  width,
  height,
  series,
  options,
}: ApexChartProps) => (
  <ReactApexChart
    type={type}
    width={width}
    series={series}
    height={height}
    options={options}
  />
);
