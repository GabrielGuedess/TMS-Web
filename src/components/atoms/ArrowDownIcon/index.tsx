import { forwardRef, type ForwardRefRenderFunction } from 'react';

import { type ArrowDownIconProps } from './types';

const ArrowDownIconRef: ForwardRefRenderFunction<
  SVGSVGElement,
  ArrowDownIconProps
> = ({ size = 10, ...props }, ref) => (
  <svg
    ref={ref}
    fill="none"
    width={size}
    height={size}
    viewBox="0 0 10 10"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fill="currentColor"
      d="M5.64284 7.69237L9.09102 4.33987L10 5.22362L5 10.0849L-8.98488e-07 5.22362L0.908973 4.33987L4.35716 7.69237L4.35716 0.0848701L5.64284 0.0848704L5.64284 7.69237Z"
    />
  </svg>
);

export const ArrowDownIcon = forwardRef(ArrowDownIconRef);
