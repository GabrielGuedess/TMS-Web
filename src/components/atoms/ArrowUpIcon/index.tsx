import { forwardRef, type ForwardRefRenderFunction } from 'react';

import { type ArrowUpIconProps } from './types';

const ArrowUpIconRef: ForwardRefRenderFunction<
  SVGSVGElement,
  ArrowUpIconProps
> = ({ size = 10, ...props }, ref) => (
  <svg
    ref={ref}
    fill="none"
    width={size}
    height={size}
    viewBox="0 0 10 10"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fill="currentColor"
      d="M4.35716 2.47737L0.908974 5.82987L5.0443e-07 4.94612L5 0.0848689L10 4.94612L9.09103 5.82987L5.64284 2.47737L5.64284 10.0849L4.35716 10.0849L4.35716 2.47737Z"
    />
  </svg>
);

export const ArrowUpIcon = forwardRef(ArrowUpIconRef);
