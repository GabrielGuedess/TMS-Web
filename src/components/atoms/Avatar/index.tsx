import Image from 'next/image';

import {
  forwardRef,
  type ElementRef,
  type ForwardRefRenderFunction,
} from 'react';

import { twMerge } from 'tailwind-merge';

import { type AvatarProps } from './types';

const AvatarRef: ForwardRefRenderFunction<
  ElementRef<typeof Image>,
  AvatarProps
> = ({ alt, src, className, ...props }, ref) => (
  <div
    className={twMerge(
      'relative w-10 h-10 select-none overflow-hidden rounded-full bg-porcelain-50 dark:bg-smoke-950 transition-all',
      className,
    )}
  >
    <Image
      ref={ref}
      src={src}
      alt={alt}
      className="select-none object-cover"
      sizes="(max-width: 768px) 100vw, 33vw"
      fill
      priority
      {...props}
    />
  </div>
);

export const Avatar = forwardRef(AvatarRef);
