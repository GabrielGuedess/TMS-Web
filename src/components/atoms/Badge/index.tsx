import { forwardRef, type ForwardRefRenderFunction } from 'react';

import { clsx } from 'clsx';
import { twMerge } from 'tailwind-merge';

import { type BadgeProps } from './types';

const BadgeRef: ForwardRefRenderFunction<HTMLDivElement, BadgeProps> = (
  { title, className, color = 'primary', variant = 'label', ...props },
  ref,
) => (
  <div className={twMerge('relative w-min', className)} {...props}>
    <div
      className={clsx('absolute left-0 top-0 size-full rounded', {
        'bg-shark-950': color === 'dark',
        'bg-cobalt-700': color === 'info',
        'bg-danger-500': color === 'danger',
        'opacity-[16%]': variant === 'label',
        'bg-primary-400': color === 'primary',
        'bg-success-500': color === 'success',
        'bg-warning-500': color === 'warning',

        'bg-dark-200 dark:bg-shark-950': color === 'secondary',
      })}
      ref={ref}
    />

    <div
      className={clsx('flex px-[0.625rem] py-[0.313rem]', {
        'text-white': variant === 'solid',

        'text-cobalt-700': color === 'info' && variant === 'label',
        'text-danger-500': color === 'danger' && variant === 'label',
        'text-primary-400': color === 'primary' && variant === 'label',
        'text-success-500': color === 'success' && variant === 'label',
        'text-warning-500': color === 'warning' && variant === 'label',
        'text-secondary-500': color === 'secondary' && variant === 'label',
        'bg-dark-200 dark:bg-shark-950':
          color === 'dark' && variant === 'label',
      })}
    >
      <span className="z-10 text-center text-[0.813rem] font-semibold capitalize">
        {title}
      </span>
    </div>
  </div>
);

export const Badge = forwardRef(BadgeRef);
