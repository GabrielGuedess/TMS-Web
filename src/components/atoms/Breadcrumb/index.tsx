import Link from 'next/link';

import { forwardRef, type ForwardRefRenderFunction } from 'react';

import { type BreadcrumbProps } from './types';

const BreadcrumbRef: ForwardRefRenderFunction<
  HTMLDivElement,
  BreadcrumbProps
> = ({ pageName, ...props }, ref) => (
  <div
    ref={ref}
    className="flex flex-col gap-3 sm:flex-row sm:items-center sm:justify-between"
    {...props}
  >
    <h2 className="text-xl font-semibold text-shark-950 transition-all dark:text-white-lilac-50 md:text-2xl">
      {pageName}
    </h2>

    <nav>
      <ol className="flex items-center gap-2">
        <li className="text-shark-950 transition-all dark:text-white-lilac-50">
          <Link
            href="/dashboard"
            className="font-medium text-shark-950 transition-all dark:text-white-lilac-50"
          >
            Dashboard
          </Link>
        </li>
        /<li className="font-medium text-primary-400">{pageName}</li>
      </ol>
    </nav>
  </div>
);

export const Breadcrumb = forwardRef(BreadcrumbRef);
