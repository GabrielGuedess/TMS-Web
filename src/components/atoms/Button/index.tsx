'use client';

import { forwardRef, type ForwardRefRenderFunction } from 'react';

import clsx from 'clsx';
import { twMerge } from 'tailwind-merge';

import { type ButtonProps } from './types';

const ButtonRef: ForwardRefRenderFunction<HTMLButtonElement, ButtonProps> = (
  {
    children,
    iconFirst,
    className,
    iconSecond,
    size = 'default',
    variant = 'solid',
    color = 'primary',
    isLoading = false,
    iconAlign = 'left',
    isDisabled = false,
    isFullWidth = false,
    ...props
  },
  ref,
) => {
  const hasBackground = variant !== 'outline';
  const hasTextColor = variant !== 'solid';

  return (
    <button
      className={twMerge(
        clsx('relative transition-transform duration-200 ease-in-out', {
          'w-full': isFullWidth,
          'opacity-50 cursor-not-allowed': isLoading || isDisabled,
          'hover:scale-105 focus:scale-105 active:scale-90':
            !isLoading && !isDisabled,
        }),
        className,
      )}
      ref={ref}
      type="button"
      disabled={isDisabled}
      {...props}
    >
      <div
        className={clsx('absolute size-full rounded-md', {
          'opacity-[16%]': variant === 'label',
          'bg-shark-950': color === 'dark' && hasBackground,
          'bg-cobalt-700': color === 'info' && hasBackground,
          'bg-danger-500': color === 'danger' && hasBackground,
          'bg-primary-400': color === 'primary' && hasBackground,
          'bg-success-500': color === 'success' && hasBackground,
          'bg-warning-500': color === 'warning' && hasBackground,

          'bg-dark-200 dark:bg-shark-950':
            color === 'secondary' && hasBackground,
          'border-dark-500 border-[1px]':
            color === 'dark' && variant === 'outline',
          'border-cobalt-700 border-[1px]':
            color === 'info' && variant === 'outline',
          'border-danger-500 border-[1px]':
            color === 'danger' && variant === 'outline',
          'border-dark-200 border-[1px]':
            color === 'secondary' && variant === 'outline',
          'border-primary-400 border-[1px]':
            color === 'primary' && variant === 'outline',
          'border-success-500 border-[1px]':
            color === 'success' && variant === 'outline',

          'border-warning-500 border-[1px]':
            color === 'warning' && variant === 'outline',
        })}
        aria-label="Button Background"
      />

      <div
        className={clsx(
          'flex h-full items-center justify-center gap-2 px-5 py-[0.625rem]',
          {
            'flex-row': iconAlign === 'left',
            'text-white': variant === 'solid',

            'flex-row-reverse': iconAlign === 'right',

            'text-shark-950': color === 'dark' && hasTextColor,
            'text-cobalt-700': color === 'info' && hasTextColor,
            'text-danger-500': color === 'danger' && hasTextColor,
            'text-primary-400': color === 'primary' && hasTextColor,
            'text-success-500': color === 'success' && hasTextColor,
            'text-warning-500': color === 'warning' && hasTextColor,
            'text-gray-900 dark:text-dark-300':
              color === 'secondary' && hasTextColor,
          },
        )}
      >
        {iconFirst && <div className="z-10">{iconFirst}</div>}

        {!!children && !isLoading && (
          <span
            className={clsx(
              'z-10 font-medium leading-[1.125rem] tracking-[0.027rem]',
              {
                'text-sm': size === 'small',
                'text-lg': size === 'large',
                'text-[0.9375rem]': size === 'default',
              },
            )}
          >
            {children}
          </span>
        )}

        {iconSecond && <div className="z-10">{iconSecond}</div>}

        {isLoading && (
          <svg
            width="18"
            height="18"
            fill="none"
            viewBox="0 0 22 22"
            aria-label="Loading"
            className="animate-spin"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              stroke="currentColor"
              d="M14.4375 3.4375C15.8767 4.10872 17.0942 5.17707 17.9468 6.51682C18.7993 7.85658 19.2515 9.41198 19.25 11C19.25 13.188 18.3808 15.2865 16.8336 16.8336C15.2865 18.3808 13.188 19.25 11 19.25C8.81197 19.25 6.71355 18.3808 5.16637 16.8336C3.6192 15.2865 2.75 13.188 2.75 11C2.74851 9.41198 3.20066 7.85658 4.05323 6.51682C4.9058 5.17707 6.12331 4.10872 7.5625 3.4375"
            />
          </svg>
        )}
      </div>
    </button>
  );
};

export const Button = forwardRef(ButtonRef);
