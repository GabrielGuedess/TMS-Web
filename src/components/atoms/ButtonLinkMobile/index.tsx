'use client';

import Link from 'next/link';
import { usePathname } from 'next/navigation';

import { forwardRef, type ForwardRefRenderFunction } from 'react';

import clsx from 'clsx';
import { twMerge } from 'tailwind-merge';

import { type ButtonLinkMobileProps } from './types';

const ButtonLinkMobileRef: ForwardRefRenderFunction<
  HTMLAnchorElement,
  ButtonLinkMobileProps
> = ({ href, title, className, ...props }, ref) => {
  const pathname = usePathname();

  return (
    <Link
      className={twMerge(
        clsx('w-full p-4 text-center text-xl transition-all', {
          'dark:bg-shark-950 bg-primary-400 text-white-lilac-50':
            pathname === href,
          'dark:hover:bg-shark-950 hover:bg-primary-400 text-comet-500 dark:text-dark-300 hover:text-white-lilac-50':
            pathname !== href,
        }),
        className,
      )}
      ref={ref}
      href={href}
      {...props}
    >
      {title}
    </Link>
  );
};

export const ButtonLinkMobile = forwardRef(ButtonLinkMobileRef);
