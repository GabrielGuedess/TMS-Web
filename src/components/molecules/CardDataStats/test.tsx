import { render } from '@testing-library/react';

import { DashboardIcon } from 'components/atoms/DashboardIcon';

import { CardDataStats } from '.';

describe('<CardDataStats />', () => {
  it('should render the card data stats correctly', () => {
    const { container } = render(
      <CardDataStats rate="12" total="123" title="Test" icon={DashboardIcon} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
