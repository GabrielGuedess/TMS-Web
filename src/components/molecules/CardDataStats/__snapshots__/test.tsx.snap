// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDataStats /> should render the card data stats correctly 1`] = `
<div
  class="glow"
  style="display: grid; --glow-top: 0px; --glow-left: 0px;"
>
  <div
    class="rounded"
    style="grid-area: 1/1/1/1;"
  >
    <div
      class="rounded border border-zumthor-100 bg-white-lilac-50 px-[1.875rem] py-6 shadow-default transition-all glow:border-glow glow:bg-glow/[.15] glow:ring-1 glow:ring-glow dark:border-shark-950 dark:bg-smoke-950"
    >
      <div
        class="flex size-[2.875rem] items-center justify-center rounded-full bg-porcelain-50 transition-all dark:bg-shark-950"
      >
        <svg
          class="transition-all glow:text-glow"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 8.52V3.98C22 2.57 21.36 2 19.77 2H15.73C14.14 2 13.5 2.57 13.5 3.98V8.51C13.5 9.93 14.14 10.49 15.73 10.49H19.77C21.36 10.5 22 9.93 22 8.52Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M22 19.77V15.73C22 14.14 21.36 13.5 19.77 13.5H15.73C14.14 13.5 13.5 14.14 13.5 15.73V19.77C13.5 21.36 14.14 22 15.73 22H19.77C21.36 22 22 21.36 22 19.77Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M10.5 8.52V3.98C10.5 2.57 9.86 2 8.27 2H4.23C2.64 2 2 2.57 2 3.98V8.51C2 9.93 2.64 10.49 4.23 10.49H8.27C9.86 10.5 10.5 9.93 10.5 8.52Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M10.5 19.77V15.73C10.5 14.14 9.86 13.5 8.27 13.5H4.23C2.64 13.5 2 14.14 2 15.73V19.77C2 21.36 2.64 22 4.23 22H8.27C9.86 22 10.5 21.36 10.5 19.77Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </div>
      <div
        class="mt-4 flex items-end justify-between"
      >
        <div>
          <h4
            class="text-[1.5rem] font-bold text-shark-900 transition-all dark:text-white"
          >
            123
          </h4>
          <span
            class="text-sm font-medium text-comet-500 transition-all dark:text-dark-300"
          >
            Test
          </span>
        </div>
        <span
          class="flex items-center gap-1 text-sm font-medium text-success-500"
        >
          12
          <svg
            class="text-success-500"
            fill="none"
            height="10"
            viewBox="0 0 10 10"
            width="10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.35716 2.47737L0.908974 5.82987L5.0443e-07 4.94612L5 0.0848689L10 4.94612L9.09103 5.82987L5.64284 2.47737L5.64284 10.0849L4.35716 10.0849L4.35716 2.47737Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    class="glow-mask rounded"
    glow="true"
    style="--glow-color: green; grid-area: 1/1/1/1; pointer-events: none; mask: radial-gradient(var(--glow-size) var(--glow-size) at calc(var(--glow-x, -99999px) - var(--glow-left, 0px))
calc(var(--glow-y, -99999px) - var(--glow-top, 0px)), #000000 1%, transparent 50%);"
  >
    <div
      class="rounded border border-zumthor-100 bg-white-lilac-50 px-[1.875rem] py-6 shadow-default transition-all glow:border-glow glow:bg-glow/[.15] glow:ring-1 glow:ring-glow dark:border-shark-950 dark:bg-smoke-950"
    >
      <div
        class="flex size-[2.875rem] items-center justify-center rounded-full bg-porcelain-50 transition-all dark:bg-shark-950"
      >
        <svg
          class="transition-all glow:text-glow"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 8.52V3.98C22 2.57 21.36 2 19.77 2H15.73C14.14 2 13.5 2.57 13.5 3.98V8.51C13.5 9.93 14.14 10.49 15.73 10.49H19.77C21.36 10.5 22 9.93 22 8.52Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M22 19.77V15.73C22 14.14 21.36 13.5 19.77 13.5H15.73C14.14 13.5 13.5 14.14 13.5 15.73V19.77C13.5 21.36 14.14 22 15.73 22H19.77C21.36 22 22 21.36 22 19.77Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M10.5 8.52V3.98C10.5 2.57 9.86 2 8.27 2H4.23C2.64 2 2 2.57 2 3.98V8.51C2 9.93 2.64 10.49 4.23 10.49H8.27C9.86 10.5 10.5 9.93 10.5 8.52Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M10.5 19.77V15.73C10.5 14.14 9.86 13.5 8.27 13.5H4.23C2.64 13.5 2 14.14 2 15.73V19.77C2 21.36 2.64 22 4.23 22H8.27C9.86 22 10.5 21.36 10.5 19.77Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </div>
      <div
        class="mt-4 flex items-end justify-between"
      >
        <div>
          <h4
            class="text-[1.5rem] font-bold text-shark-900 transition-all dark:text-white"
          >
            123
          </h4>
          <span
            class="text-sm font-medium text-comet-500 transition-all dark:text-dark-300"
          >
            Test
          </span>
        </div>
        <span
          class="flex items-center gap-1 text-sm font-medium text-success-500"
        >
          12
          <svg
            class="text-success-500"
            fill="none"
            height="10"
            viewBox="0 0 10 10"
            width="10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.35716 2.47737L0.908974 5.82987L5.0443e-07 4.94612L5 0.0848689L10 4.94612L9.09103 5.82987L5.64284 2.47737L5.64284 10.0849L4.35716 10.0849L4.35716 2.47737Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
